backend/fms_core/tests/test_services/sample_test.py

```diff
@@ -731,4 +734,47 @@ def test_pool_submitted_samples(self):
             else:
                 self.assertEqual(derived_sample.biosample.individual, self.test_individuals[1])
                 self.assertEqual(derived_by_sample_pool_.volume_ratio, Decimal("0.75"))
-                self.assertEqual(derived_sample.library, self.test_libraries[5])
\ No newline at end of file
+                self.assertEqual(derived_sample.library, self.test_libraries[5])
+
+    def test_test_and_update_volume_ratio(self):
+        samples: list[dict[str, Any]] = [{ "alias": "A" }, { "alias": "B" }, { "alias": "C" }, { "alias": "D" }]
```

Comment:
Test using a number of sample that do not split exactly to see how it handles the rounding error issue.